test/integration-test.ts

```diff
@@ -535,6 +539,251 @@ describe('integration', function() {
+            }
+          }
+        },
+        'app/components/hello/index.hbs',
+        { line: 0, character: 3 }
+      );
+      expect(result).toMatchSnapshot();
+    });
+
+    it('support dummy addon definition:script', async () => {
+      const result = await getResult(
+        DefinitionRequest.type,
+        connection,
+        {
+          node_modules: {
+            provider: {
+              lib: {
+                'langserver.js': `
+                  module.exports.onDefinition = function(root) { 
+                    let filePath = require("path").join(__dirname, "./../../../app/components/hello/index.js");
+                    return [ {
+                      "range": {
+                        "end": {
+                          "character": 0,
+                          "line": 0,
+                        },
+                        "start": {
+                          "character": 0,
+                          "line": 0,
+                        }
+                      },
+                      "uri": "file://" + filePath.split(':').pop()
+                    } ]; 
+                  }
+                `
+              },
+              'package.json': JSON.stringify({
+                name: 'provider',
+                'ember-language-server': {
+                  entry: './lib/langserver',
+                  capabilities: {
+                    definitionProvider: true
+                  }
+                }
+              })
+            }
+          },
+          'package.json': JSON.stringify({
+            dependencies: {
+              provider: '*'
+            }
+          }),
+          app: {
+            components: {
+              hello: {
+                'index.js': 'var n = "foo";'
+              }
+            }
+          }
+        },
+        'app/components/hello/index.js',
+        { line: 0, character: 10 }
+      );
+      expect(result).toMatchSnapshot();
+    });
+  });
+
+  describe('Able to provide API:Reference', () => {
+    it('support dummy addon reference:template', async () => {
+      const result = await getResult(
+        ReferencesRequest.type,
+        connection,
+        {
+          node_modules: {
+            provider: {
+              lib: {
+                'langserver.js': `
+                  module.exports.onReference = function(root) { 
+                    let filePath = require("path").join(__dirname, "./../../../app/components/hello/index.hbs");
+                    return [ { uri: 'file://' + filePath.split(':').pop(), range: { start: { line: 0, character: 0 }, end: { line: 0, character: 0 } } } ];
+                  }
+                `
+              },
+              'package.json': JSON.stringify({
+                name: 'provider',
+                'ember-language-server': {
+                  entry: './lib/langserver',
+                  capabilities: {
+                    referencesProvider: true
+                  }
+                }
+              })
```

Comment:
Is this all that's needed for a plugin to be accepted? Any it has to be in the project's dependencies?

Reply:
Correct, plugin looks like dependency inside project.
Dependency has special section inside `package.json` and handler file.
It can be ember addon as well (with section inside `package.json`)

Reply:
simple autocomplete addon file structure

`lib/langserver.js`
```js
module.exports.onComplete = function(root, { ...params }) { 
  return [{label: "this.name"}];
}
```

`package.json`
```json
{
  "name": "dummy-provider",
   "ember-language-server": {
      "entry": "./lib/langserver",
      "capabilities": {
        "completionProvider": true
      }
  }
}
```